### PR TITLE
get coin id from platforms

### PR DESCRIPTION
**Pull request by testing-life**
## What was done
Another dictionary is created allowing to look for coin id by address, rather than just name 
image

#### After

### CODE REVIEW

**Comment by blink-km** [comment]
@hiaux0 @giladt Could someone review this PR please?

File: src/services/TokenService.ts [hunk 1]
```diff
@@ -71,7 +71,8 @@ export class TokenService {
 
   public async initialize(): Promise<TokenListMap> {
     this.geckoCoinInfo = new Map<string, string>();
-    const uri = `https://pro-api.coingecko.com/api/v3/coins/list?x_cg_pro_api_key=${process.env.COINGECKO_API_KEY}`;
+    this.coinAddressIdDictionary = new Map<string, string>();
```
Comment (hiaux0) [inline src/services/TokenService.ts:74]: There is an `Address` type

File: src/services/TokenService.ts [hunk 2]
```diff
@@ -71,7 +71,8 @@ export class TokenService {
...
+    const uri = `https://pro-api.coingecko.com/api/v3/coins/list?include_platform=true&x_cg_pro_api_key=${process.env.COINGECKO_API_KEY}`;
```
Comment (hiaux0) [inline src/services/TokenService.ts:75]: Would have expected to have some words about `include_platform` in the initial comment pointing to their api docs
https://www.coingecko.com/en/api/documentation

File: src/services/TokenService.ts [hunk 3]
```diff
@@ -80,17 +81,23 @@ export class TokenService {
...
       });
     TimingService.end("get geckoCoinInfo");
-
     return this.tokenLists = await this.tokenListService.fetchLists();
   }
 
   private geckoCoinInfo: Map<string, string>;
+  private coinAddressIdDictionary: Map<string, string>;
 
+  private coinMapping(tokenInfo: ITokenInfo){
```
Comment (hiaux0) [inline src/services/TokenService.ts:94]: return type missing and formatting `){`

File: src/services/TokenService.ts [hunk 4]
```diff
@@ -100,8 +107,8 @@ export class TokenService {
     return `${name.toLowerCase()}_${symbol.toLowerCase()}`;
   }
 
-  private getTokenGeckoId(name: string, symbol: string): string {
-    const id = this.geckoCoinInfo.get(this.getTokenGeckoMapKey(name, symbol));
+  private getTokenGeckoId(name: string, symbol: string, address:string = null): string {
```
Comment (hiaux0) [inline src/services/TokenService.ts:110]: with just type `:string` I would expect the default to be `""` and not `null`

File: src/services/TokenService.ts (hunk 1 above)
Comment (hiaux0) [reply src/services/TokenService.ts:74]: But okay, the above Map also just has `string`

**Review by tamagosante — APPROVED** [review]
### Fncality 
:heavy_check_mark: 
image


### Code

**Pull request merged**